AudioPlayer/src/main/kotlin/com.github.diamondminer88.plugins/AudioPlayer.kt

```diff
@@ -29,246 +29,404 @@ import com.google.android.material.card.MaterialCardView
-					override fun onProgressChanged(seekBar: SeekBar, progress: Int, fromUser: Boolean) {
-						if (!fromUser) return
-						if (!isPrepared) {
-							seekBar.progress = prevProgress
-							return
-						}
-						prevProgress = progress
-						currentPlayer!!.seekTo((progress.div(500f) * duration).toInt())
-						progressView.text =
-							"${msToTime(currentPlayer!!.currentPosition.toLong())} / ${msToTime(duration)}"
-					}
-				})
-
-				buttonView.setOnClickListener {
-					playing = !playing
-
-					if (!isPrepared && !preparing) {
-						preparing = true
-						Utils.mainThread.post { buttonView.background = null }
-
-						currentPlayer?.release()
-						currentPlayerUnsubscribe?.invoke()
-						onPauseListener = null
-						var url = messageAttachment.url
-						currentPlayer = MediaPlayer()
-
-						Utils.threadPool.execute {
-
-							if (messageAttachment.filename.endsWith(".ogg")) {
-								var file = File(ctx.cacheDir, "audio.ogg")
-								file.deleteOnExit()
-								Http.simpleDownload(url, file)
-								url = file.absolutePath
-							}
-
-							currentPlayer?.apply {
-								setDataSource(url)
-								setOnPreparedListener {
-									seekTo((sliderView.progress.div(500f) * duration).toInt())
-									Utils.mainThread.post { updatePlaying() }
-									duration = it.duration.toLong()
-								}
-								setOnCompletionListener { player ->
-									playing = false
-									player.seekTo(0)
-									Utils.mainThread.post { buttonView.background = rewindIcon }
-								}
-								currentPlayerUnsubscribe = {
-									playing = false
-									Utils.mainThread.post { updatePlaying() }
-								}
-								onPauseListener = {
-									playing = false
-									Utils.mainThread.post { updatePlaying() }
-								}
-								prepare()
-								isPrepared = true
-								preparing = false
-							}
-						}
-					} else {
-						updatePlaying()
-					}
-				}
-
-				addView(buttonView)
-				addView(progressView)
-				addView(sliderView)
-			})
-		}
-
-		patcher.after<StoreMessages>("handleChannelSelected", Long::class.javaPrimitiveType!!) {
-			currentPlayerUnsubscribe?.invoke()
-			currentPlayerUnsubscribe = null
-			onPauseListener = null
-		}
-
-		patcher.after<AppActivity>("onCreate", Bundle::class.java) {
-			currentPlayerUnsubscribe?.invoke()
-			currentPlayerUnsubscribe = null
-			onPauseListener = null
-		}
-
-		patcher.after<AppActivity>("onPause") {
-			onPauseListener?.invoke()
-		}
-	}
-
-	override fun stop(context: Context) {
-		patcher.unpatchAll()
-	}
-}
+    fun deleteOggCacheFiles(cacheDir: File, oggCacheMap: Map<String, File>) {
+        val oggFiles = cacheDir.listFiles { file ->
+            file.parentFile == cacheDir &&
+            file.name.matches(Regex("^audio_-?\\d+\\.ogg$"))
+        } ?: emptyArray()
+        val filesToDelete = oggFiles.filter { file -> !oggCacheMap.values.contains(file) }
+        for (file in filesToDelete) {
+            file.delete()
+        }
+    }
```

Comment:
You should store the cached ogg files in a subdir of the cache dir. This makes deleting it very easy with a `File#
deleteRecursively()` on `start()`. Read my other review comments about wiping cache.